Update Dockerfile to solve requirements issue

grantfitzsimmons opened the pull request:
Solves the `requirements.txt` issue that I've been dealing with (Issue #3793). Need to have a developer check this out.

grantfitzsimmons committed:
Update Dockerfile to solve requirements issue

Solves the `requirements.txt` issue that I've been dealing with. Need to have a developer check this out.

grantfitzsimmons commented:
When cloning a record with a `uniqueIdentifier` value, it is preserved during the clone process. This cannot be reproduced on `v7.9-dev`. When I delete the unique identifier manually, it saves correctly.

When saving, I get this crash:

```
TransactionManagementError at /api/specify/collectionobject/ An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block. Request Method: POST Request URL: http://localhost/api/specify/collectionobject/ Django Version: 3.2.15 Python Executable: /opt/specify7/ve/bin/python Python Version: 3.8.10 Python Path: ['/opt/specify7', '/opt/specify7', '/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '/opt/specify7/ve/lib/python3.8/site-packages'] Server time: Mon, 17 Jul 2023 11:20:47 -0500
```

<img width="942" alt="image" src="https://github.com/specify/specify7/assets/37256050/f079e6d9-3718-4007-8cbb-5116893a419a">

[Specify 7 Crash Report - 2023-07-17T16_22_04.631Z.txt](https://github.com/specify/specify7/files/12071703/Specify.7.Crash.Report.-.2023-07-17T16_22_04.631Z.txt)

grantfitzsimmons committed:
Merge branch 'v7.9-dev' into issue-3793

maxpatiiuk reviewed (approved):
this looks good

though while we are at this I would suggest updating other dependencies too.

i.e python 3.8 is end of life in a year. django 3.2 is end of life next year too

grantfitzsimmons committed:
Merge branch 'v7.9-dev' into issue-3793

melton-jason committed:
Merge branch 'v7.9-dev' into issue-3793

melton-jason committed:
Lint code with ESLint and Prettier

Triggered by fc82601f508e2ade6a70acd609f916a84b331a0a on branch refs/heads/issue-3793
